test/core/compression/stream_compression_fuzzer.cc

```diff
@@ -0,0 +1,33 @@
+#include <grpc/grpc.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "src/core/lib/compression/stream_compression.h"
+#include "src/core/lib/security/credentials/credentials.h"
+#include "test/core/util/memory_counters.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  grpc_core::testing::LeakDetector leak_detector(true);
+  grpc_init();
+  grpc_test_only_control_plane_credentials_force_init();
+  auto* context = grpc_stream_compression_context_create(
+      GRPC_STREAM_COMPRESSION_GZIP_COMPRESS);
+  grpc_slice_buffer input_buffer;
+  grpc_slice_buffer_init(&input_buffer);
+  grpc_slice_buffer_add(
+      &input_buffer,
+      grpc_slice_from_copied_buffer(reinterpret_cast<const char*>(data), size));
+  grpc_slice_buffer output_buffer;
+  grpc_slice_buffer_init(&output_buffer);
+
+  grpc_stream_compress(context, &input_buffer, &output_buffer, nullptr,
+                       (~(size_t)0), GRPC_STREAM_COMPRESSION_FLUSH_SYNC);
```

Comment:
Setting the max_output_size to 0 seems like it would skip most of the functionality we're trying to test here.

https://github.com/grpc/grpc/blob/0c8bedca3334656a02dc39cf248adf5749a11d60/src/core/lib/compression/stream_compression_gzip.cc#L49

Reply:
Sorry, it's not actually being set to 0, but ~0 (`std::numeric_limits<size_t>::max()`, but portable). Replaced with `(size_t)-1` to make it more clear what's happening.